lib/src/simple_moment_base.dart

```diff
@@ -9,25 +11,128 @@ class Moment {
   DateTime _date;
   static ILocaleData _globalLocale = new LocaleEn();
   ILocaleData _locale;
+  static int monthPerQuarter = 3;
 
   Moment.now() {
     _date = new DateTime.now();
   }
 
+  /// Create a moment instance from Date
+  ///
+  /// @param date The date
   Moment.fromDate(DateTime date) {
     _date = date;
   }
 
+  /// Create a date from milliseconds
+  ///
+  /// @param millisecondsSinceEpoch
   Moment.fromMillisecondsSinceEpoch(int millisecondsSinceEpoch,
       {bool isUtc: false}) {
     _date = new DateTime.fromMillisecondsSinceEpoch(millisecondsSinceEpoch,
         isUtc: isUtc);
   }
 
+  /// Converts a Date String to an actual DateTime object
+  ///
+  /// @param date The date String
+  /// @returns a moment instance
   static Moment parse(String date) {
     return new Moment.fromDate(DateTime.parse(date));
   }
 
+  Moment.fromDateTime(this._date);
+
+  ///
+  /// Adds time to the current date
+  ///
+  Moment add({
+    int years: 0,
+    int quarters: 0,
+    int months: 0,
+    int weeks: 0,
+    int days: 0,
+    int hours: 0,
+    int minutes: 0,
+    int seconds: 0,
+    int milliseconds: 0,
+    int microseconds: 0,
+  }) {
+    var dateTime = DateTime(
+      _date.year + years,
+      _date.month + months + quarters * monthPerQuarter,
+      _date.day + days + weeks * DateTime.daysPerWeek,
+      _date.hour + hours,
+      _date.minute + minutes,
+      _date.second + seconds,
+      _date.millisecond + milliseconds,
+      _date.microsecond + microseconds,
+    );
+    return Moment.fromDateTime(dateTime);
+  }
+
+  ///
+  /// Substracts time to the current date
+  ///
+  Moment subtract({
+    int years: 0,
+    int quarters: 0,
+    int months: 0,
+    int weeks: 0,
+    int days: 0,
+    int hours: 0,
+    int minutes: 0,
+    int seconds: 0,
+    int milliseconds: 0,
+    int microseconds: 0,
+  }) {
+    var dateTime = DateTime(
+      _date.year - years,
+      _date.month - months - quarters * monthPerQuarter,
```

Comment:
Please add tests for adding and subtracting with months and days, too, since you're doing some calculations there. 😄